src/omnisharp/options.ts

```diff
@@ -134,6 +119,28 @@ export class Options {
         );
     }
 
+    public static getExcludedPaths(vscode: vscode): string[] {
+        let excludePaths: string[] = [];
+
+        let workspaceConfig = vscode.workspace.getConfiguration();
+        if (!workspaceConfig) {
+            return excludePaths;
+        }
+
+        let excludeFilesOption = workspaceConfig.get<{ [i: string]: boolean }>('files.exclude');
```

Comment:
Out of curiosity does VSCode pre-fill node_modules in this list?

Reply:
unfortunately it wasn't set for the `dotnet new razor` project I created to test with.

Reply:
It might make sense for us to auto-include `node_modules` and whatever Unity's package folder is if it doesn't already exist.

Reply:
Maybe this should also include `search.exclude`.. hmm..

Reply:
Good news is that Unity seems to create a .vscode/settings.json which has an extensive `files.exclude` defined

Reply:
Updated to include `search.exclude` paths which by [default include `node_modules`](https://github.com/microsoft/vscode/blob/f68b1bd2a3045f4ca2153e0957a1fda168d5e467/src/vs/workbench/contrib/search/browser/search.contribution.ts#L687). Unity's settings.json includes the Library/PackageCache in its `files.exclude` so I think we are covered from these scenarios.